### PR TITLE
Lookup api query changed to improve performance

### PR DESCRIPTION
**Pull request by rahulrkhapre**
This PR includes the query change for lookup API. It was observed that lookup API performance was very slow when number of records are more than 18K, this change will improve the performance of the lookup API

### CODE REVIEW

File: backend/src/main/java/org/eclipse/tractusx/semantics/registry/repository/ShellRepository.java [hunk 1]
```diff
@@ -53,13 +53,11 @@ public interface ShellRepository extends PagingAndSortingRepository<Shell, UUID>
...
     @Query(
-            "select s.id_external from shell s where s.id in (" +
-                    "select si.fk_shell_id from shell_identifier si " +
+            "select s.id_external from shell s join shell_identifier si on s.id = si.fk_shell_id " +
                     "where concat(si.namespace,si.identifier) in (:keyValueCombinations) " +
-                    "AND ((si.external_subject_id is not null and si.external_subject_id = :tenantId) or :tenantId = s.tenant_id) " +
-                    "group by si.fk_shell_id " +
-                    "having count(*) = :keyValueCombinationsSize " +
-            ")"
+                    "AND ((si.external_subject_id is not null and si.external_subject_id = :tenantId) " +
```
Comment (shijinrajbosch) [inline backend/src/main/java/org/eclipse/tractusx/semantics/registry/repository/ShellRepository.java:58]: is this redundant condition - `si.external_subject_id is not null ` . Can it work with this condition alone -  `si.external_subject_id = :tenantId` ?

Comment (rahulrkhapre) [reply backend/src/main/java/org/eclipse/tractusx/semantics/registry/repository/ShellRepository.java:58]: This is an old query only and we have removed the IN condition and replaced it with JOIN.

Comment (shijinrajbosch) [reply backend/src/main/java/org/eclipse/tractusx/semantics/registry/repository/ShellRepository.java:58]: Thanks for the clarification @rahulrkhapre . The performance will improve if we remove redundant condition.

**Review by tunacicek — APPROVED** [review]
LGTM

**Pull request merged**